thunderbolt/s3_client.py

```diff
@@ -0,0 +1,56 @@
+import os
+import pickle
+from typing import List, Dict, Any
+
+import boto3
+from boto3 import Session
+from tqdm import tqdm
+
+
+class S3Client:
```

Comment:
Great refactoring!
It would be better if these clients are putted into the structured module like 'thunderbolt/clients'.
Of course, you can even use abstract base class if you like.

Reply:
Sounds good! I will try to correspond easily :)